processing/src/main/java/org/apache/druid/java/util/emitter/service/ServiceMetricEvent.java

```diff
@@ -197,7 +198,7 @@ public ServiceMetricEvent build(ImmutableMap<String, String> serviceDimensions)
       return new ServiceMetricEvent(
           createdTime,
           serviceDimensions,
-          userDims,
+          new HashMap<>(userDims),
```

Comment:
I see that `StreamAppenderatorTest#testQueryByIntervals` fails if we directly pass `userDims`.

My best rationale so far is that this change is only needed because the test uses `StubServiceEmitter`, hence the previously stored metrics don't actually get emitted anywhere, and the previously stored metrics end up getting mutated because they use the same `userDims`.

@findingrish Can you confirm if this is the reason, and this is done only for test purposes, and directly passing `userDims` should work for any real world scenarios?

Please let me know if I'm missing some other aspect of this change. Appreciate your inputs, thanks!

Reply:
> Can you confirm if this is the reason, and this is done only for test purposes

Yes, this was only done for test purpose and directly passing `userDims` should work for real world scenarios.

Reply:
Yes, let's change the test rather than do a needless copy in production code.